UnicornRender/video/include/unicorn/video/Transform.hpp

```diff
@@ -0,0 +1,234 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#ifndef UNICORN_VIDEO_TRANSFORM_HPP
+#define UNICORN_VIDEO_TRANSFORM_HPP
+
+#include <unicorn/utility/SharedMacros.hpp>
+
+#define GLM_FORCE_RADIANS
+#include <glm/glm.hpp>
+#include <glm/gtc/quaternion.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+
+namespace unicorn
+{
+namespace video
+{
+/**
+ * @brief Base class for every object that can translate or rotate
+ */
+class Transform
+{
+public:
+    /** @brief Default constructor */
+    Transform();
+
+    /** @brief Default destructor */
+    virtual ~Transform() = default;
+
+    /**
+     * @brief Sets look at provided direction
+     *
+     * @param[in] direction new direction vector
+     */
+    UNICORN_EXPORT void LookAtDirection(glm::vec3 direction);
+
+    /**
+     * @brief Sets look at provided direction
+     *
+     * @param[in] direction new direction vector
+     * @param[in] upVector new up vector
+     */
+    UNICORN_EXPORT void LookAtDirection(glm::vec3 direction, glm::vec3 upVector);
+
+    /**
+     * @brief Sets up vector
+     *
+     * @param[in] upVector new up vector
+     */
+    UNICORN_EXPORT void SetUp(glm::vec3 upVector);
+
+    /**
+     * @brief      Sets the world coordinates
+     *
+     * @param[in]  x world axis
+     * @param[in]  y world axis
+     * @param[in]  z world axis
+     */
+    UNICORN_EXPORT void SetWorldCoordinates(glm::vec3 x, glm::vec3 y, glm::vec3 z);
+
+    /**
+     * @brief Translates object
+     *
+     * @param[in]  translate  The translate
+     */
+    UNICORN_EXPORT void Translate(glm::vec3 translate);
+
+    /**
+     * @brief Sets translate
+     *
+     * @param[in]  translate new translate
+     */
+    UNICORN_EXPORT void SetTranslate(glm::vec3 translate);
+
+    /**
+     * @brief Returns direction
+     *
+     * @return direction
+     */
+    UNICORN_EXPORT glm::vec3 GetDirection() const;
+
+    /**
+     * @brief Return right vector
```

Comment:
What is `right vector`? Is it just axis like `forward-right-up`?

Reply:
Yes, this is world vectors multiplied by camera orientation.